### PR TITLE
Add synchronous download and authentication support to downloader integration

### PR DESCRIPTION
**Pull request by robin-thoni**
## Proposed change

Add support to `downloader` integration for:
- Synchronous download
- Authentication (Basic and Digest)

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/61747
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: https://community.home-assistant.io/t/downloader-integration-lacks-authentication-options/344244

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @robin-thoni, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#20689).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: source/_integrations/downloader.markdown [hunk 1]
```diff
@@ -35,12 +35,16 @@ Go to the "Developer Tools", then to "Call Service", and choose `downloader/down
...
+| Service data attribute | Optional | Description                                                                   |
+|------------------------|----------|-------------------------------------------------------------------------------|
+| `url`                  | no       | The URL of the file to download.                                              |
+| `subdir`               | yes      | Download into subdirectory of **download_dir**                                |
+| `filename`             | yes      | Determine the filename.                                                       |
+| `overwrite`            | yes      | Whether to overwrite the file or not, defaults to `false`.                    |
+| `async`                | yes      | Whether to download in background without waiting or not, defaults to `true`. |
+| `auth_type`            | yes      | Authentication type: `none`, `basic` or `digest`, defaults to `None`.         |
+| `username`             | yes      | Username for authentication. Only required if **auth_type** is not `None`.    |
+| `password`             | yes      | Password for authentication. Only required if **auth_type** is not `None`.    |
```
Comment (deviantintegral) [inline source/_integrations/downloader.markdown:47]: I'm assuming this should be lowercase, unless this is actually referring to python's `None` type?

```suggestion
| `auth_type`            | yes      | Authentication type: `none`, `basic` or `digest`, defaults to `none`.         |
| `username`             | yes      | Username for authentication. Only required if **auth_type** is not `none`.    |
| `password`             | yes      | Password for authentication. Only required if **auth_type** is not `none`.    |
```

**Review by deviantintegral — COMMENTED** [review]
I haven't tested the related PR. Only one small suggestion on capitalization, but otherwise looks good!

**Comment by frenck** [comment]
Closing PR as the parent PR has been closed.

**Pull request closed without merging**